src/goto-programs/slice_global_inits.cpp

```diff
@@ -32,8 +35,8 @@ void slice_global_inits(goto_modelt &goto_model)
   const irep_idt entry_point=goto_functionst::entry_point();
   goto_functionst &goto_functions=goto_model.goto_functions;
 
-  if(!goto_functions.function_map.count(entry_point))
-    throw "entry point not found";
+  if(goto_functions.function_map.count(entry_point) == 0)
+    throw incorrect_goto_program_exceptiont("entry point not found");
 
```

Comment:
I'd say it's misleading to throw this exception. A goto program without entry point is in no way malformed or incorrect.

Reply:
Ok, I think that `invalid_user_input` would be misleading as well, so I've added `user_input_error` exception (locally in `slice_global_inits` until use elsewhere is required).